Configure SyncTarget workspace and update APIBinding documentation

Roming22 opened the pull request:
The process to bind to an APIExport now requires the right RBAC to be
granted to the user(s).

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

xinredhat commented on gitops/README.md:
```diff
@@ -173,6 +173,39 @@ spec:
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: apibinding-kubernetes
```
this ClusterRole name doesn't match with `apibinding:kubernetes` defined in ClusterRole.

Roming22 replied:
Fixed.

Roming22 commented:
@bnallapeta Please have a look and merge if you're OK with the changes.

Roming22 commented:
> In my opinion when we create the workspace for the registration of the workload clusters we should create the clusterrole and clusterrolebinding you have documented here. In the documentation (should also be in the operational manual) we should instruct users how they can use kustomize and overlays to restrict access to the workload clusters to a smaller group than system:authenticated

I agree that it would make the UX better. However, I'm a bit wary about having the default security setting be "opened to all".

fgiloux commented:
> I agree that it would make the UX better. However, I'm a bit wary about having the default security setting be "opened to all".

Obfuscation is usually not seen as security.
Moreover this gives the opportunity to get feedback on how security can get improved, which is not possible when things are hidden.

bnallapeta commented on gitops/README.md:
```diff
@@ -173,6 +173,39 @@ spec:
+    resources:
+      - apiexports
+    resourceNames:
+      - kubernetes # value of the spec/indentity/secretRef/name field of the APIExport
```
Do you mean 'identity'?

Roming22 replied:
Fixed

bnallapeta commented:
@Roming22 
1. There's a conflict currently.
2. I don't remember seeing so many files earlier. Is there another commit here by mistake?

Roming22 commented:
@bnallapeta 

> 1. Fixed
> 2. Originally, I was just updating the docs. Frederic suggested automating the process. This required some changes to make it customizable. I've updated the commit message accordingly.

Roming22 commented:
@bnallapeta I've processed your latest comment, let me know if that's not what you were expecting to see.

fgiloux commented on images/kcp-registrar/register.sh:
```diff
@@ -20,16 +20,81 @@ set -o pipefail
+        Can be set through \$KCP_SYNC_TAG.
+
+Optional arguments:
+    -w, --workspace-dir WORKSPACE_DIR
```
You need to be careful with such changes. Besides the fact that it has nothing to do with the stated PR goal and that it is disputable to see that as an "improvement" this is technically breaking the API.
DATA_DIR is referenced in multiple places:
- docs/kcp-registration.md
- gitops/sre/.tekton/kcp-registration.yaml
- gitops/pac/.tekton/kcp-registration.yaml
- DEVELOPMENT.md

fgiloux commented on images/kcp-registrar/register.sh:
```diff
@@ -20,16 +20,81 @@ set -o pipefail
+        Display this message.
+
+Example:
+    KCP_SYNC_TAG='v0.6.0-alpha0' %s --kcp-org 'root:my_org' --kcp-workspace 'my_workspace' --workspace_dir /path/to/my_dir
```
we should use the release image for the example

Roming22 replied:
Done

fgiloux commented on images/kcp-registrar/register.sh:
```diff
@@ -132,21 +199,38 @@ register() {
-    switch_org
-    printf "Switching to workspace %s\n" "${KCP_WORKSPACE}"
-    switch_ws
+set_kcp_workspace() {
```
I am really not convinced by the name of the function and the log message. What the function is doing is applying a set of manifests. These manifests:
- currently are setting up RBAC
- aiming to configure access to a compute pool

Roming22 replied:
While at the moment it's only granting access to the APIExport, it can actually configure any resource in the kcp workspace if the manifests are added to the `environment/kcp` directory.

If you have an idea for a better name, share it with me and I'll update it.

fgiloux replied:
proposal: configure_compute_ws()

Roming22 commented:
@fgiloux @bnallapeta Is there anything holding the merge of this PR?

fgiloux commented:
> Is there anything holding the merge of this PR?

@Roming22, we would have quicker iterations if we keep the PR scope focused on 1 thing.
As a result I have changed the title of the PR. Feel free to amend if you think that it could be better described.
